mono/metadata/threads.c

```diff
@@ -365,6 +368,29 @@ mono_thread_set_interruption_requested (MonoInternalThread *thread)
 {
 	//always force when the current thread is doing it to itself.
 	gboolean sync = thread == mono_thread_internal_current ();
+	/* Normally synchronous interruptions can bypass abort protection. */
+	return mono_thread_set_interruption_requested_flags (thread, sync);
```

Comment:
What about renaming `mono_thread_set_interruption_requested` to follow `mono_thread_set_self_interruption_respect_abort_prot`? 

This would allow us to get rid of the weird `_flags` variant.